src/lib/uTop.ml

```diff
@@ -32,8 +38,26 @@ let version = "%%VERSION%%"
    | History                                                         |
    +-----------------------------------------------------------------+ *)
 
+(** create a directory but doesn't raise an exception if the directory already exist *)
+let mkdir_safe dir perm =
+    try Unix.mkdir dir perm with Unix.Unix_error (Unix.EEXIST, _, _) -> ()
+
+(** create a directory, and create parent if doesn't exist *)
+let mkdir dir perm =
```

Comment:
This has still confusing and/or unconventional indentation. It should be more like the code below. In particular, you want to avoid to increase indentation after `let .. in` because that would force your code quickly to the right. There also appears to be a mix of how much indentation you use for each level.

```
let mkdir dir perm =
  let rec p_mkdir dir =
    let p_name = Filename.dirname dir in
    if p_name <> "/" && p_name <> "." then p_mkdir p_name;
    mkdir_safe dir perm
  in
  p_mkdir dir

let history = LTerm_history.create []

let history_file_name =
  ref
    (Option.map
       (fun cache_dir -> Filename.concat cache_dir "history")
       Project_dirs.cache_dir)

let () =
  Option.iter
    (fun cache_file ->
      let dir = Filename.dirname cache_file in
      mkdir dir 0o755)
    !history_file_name

```